lib/player.js

```diff
@@ -3018,13 +3018,38 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
         return;
       }
 
+      let interstitialStartTime = null;
+      let interstitialEndTime = null;
+      const interstitialValues = [];
+
       for (const cue of track.activeCues) {
         this.dispatchMetadataEvent_(cue.startTime, cue.endTime,
             cue.type, cue.value);
 
         if (this.adManager_) {
           this.adManager_.onCueMetadataChange(cue.value);
         }
+        if (cue.type == 'com.apple.quicktime.HLS') {
+          interstitialStartTime = cue.startTime;
+          interstitialEndTime = cue.endTime;
+          interstitialValues.push(cue.value);
```

Comment:
I have tried a VOD with multiple interstials at the same time and it only returns the first one. I think that in case they use the same time the idea is to use `X-ASSET-LIST`

Reply:
If you have a better suggestion, I'm open to implementing it.

Reply:
Done!